packages/pn-personafisica-webapp/src/components/Contacts/SpecialContacts.tsx

```diff
@@ -181,61 +132,6 @@ const SpecialContacts: React.FC<Props> = ({ digitalAddresses }) => {
     formik.values.addressType === ChannelType.PEC ? 'legal-contacts' : 'courtesy-contacts';
   const contactType = formik.values.addressType.toLowerCase();
 
-  const renderOption = (props: any, option: Party) => (
-    <MenuItem {...props} value={option.id} key={option.id}>
-      <DropDownPartyMenuItem name={option.name} />
-    </MenuItem>
-  );
-
-  const getOptionLabel = (option: Party) => option.name || '';
-
-  // handling of search string for sender
-  const entitySearchLabel: string = `${t('special-contacts.sender', {
-    ns: 'recapiti',
-  })}${searchStringLimitReachedText(senderInputValue)}`;
-
-  const handleChangeTouched = async (e: ChangeEvent) => {
-    formik.handleChange(e);
-    await formik.setFieldTouched(e.target.id, true, false);
-  };
-
-  const senderChangeHandler = async (_: any, newValue: Party | null) => {
-    await formik.setFieldTouched('sender', true, false);
-    await formik.setFieldValue('sender', newValue);
-    setSenderInputValue(newValue?.name ?? '');
-    if (newValue && addresses[newValue.id]) {
-      const alreadyExists =
-        addresses[newValue.id].findIndex((a) => a.channelType === formik.values.addressType) > -1;
-      setAlreadyExistsMessage(
-        alreadyExists
-          ? t(`special-contacts.${contactType}-already-exists`, {
-              ns: 'recapiti',
-            })
-          : ''
-      );
-      return;
-    }
-    setAlreadyExistsMessage('');
-  };
-
-  const addressTypeChangeHandler = async (e: ChangeEvent<HTMLInputElement>) => {
-    await formik.setFieldValue('s_value', '');
-    formik.handleChange(e);
-    if (formik.values.sender && addresses[formik.values.sender.id]) {
-      const alreadyExists =
-        addresses[formik.values.sender.id].findIndex((a) => a.channelType === e.target.value) > -1;
-      setAlreadyExistsMessage(
-        alreadyExists
-          ? t(`special-contacts.${e.target.value.toLowerCase()}-already-exists`, {
-              ns: 'recapiti',
-            })
-          : ''
-      );
-      return;
-    }
-    setAlreadyExistsMessage('');
-  };
-
   const sendSuccessEvent = (type: ChannelType) => {
```

Comment:
if we manage the creation of the contact into the PecContactItem and so on, we don't need this code anymore because it is already into the components above.

also handleCodeVerification, handleAddressUpdateError and the first useEffect can be removed.

the second useEffect can be removed because it is in the addSpecialContactDialog